qiling/os/windows/dlls/kernel32/tlhelp32.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+#
+# Cross Platform and Multi Architecture Advanced Binary Emulation Framework
+# Built on top of Unicorn emulator (www.unicorn-engine.org)
+
+import struct
+import time
+from qiling.os.windows.const import *
+from qiling.os.fncc import *
+from qiling.os.windows.fncc import *
+from qiling.os.windows.utils import *
+from qiling.os.memory import align
+from qiling.os.windows.thread import *
+from qiling.os.windows.handle import *
+from qiling.exception import *
+
+COUNTER_PROCESS32_NEXT = 0
+MAX_PROCESS32_NEXT = 3
+
+
+# HANDLE CreateToolhelp32Snapshot(
+#   DWORD dwFlags,
+#   DWORD th32ProcessID
+# );
+@winapi(cc=STDCALL, params={
+    "dwFlags": DWORD,
+    "th32ProcessID": DWORD
+
+})
+def hook_CreateToolhelp32Snapshot(ql, address, params):
+    # TODO thinking about implementing an handler, gonna see if is really necessary
+    flag = params["dwFlags"]
+    if flag == TH32CS_SNAPPROCESS:
+        ql.dprint("[.] We probably found a Ransomware!")
+    else:
+        raise QlErrorNotImplemented("[!] API not implemented")
+    return 0xD10C
+
+
+# BOOL Process32FirstW(
+#   HANDLE            hSnapshot,
+#   LPPROCESSENTRY32W lppe
+# );
+@winapi(cc=STDCALL, params={
+    "hSnapshot": HANDLE,
+    "lppe": POINTER
+
+})
+def hook_Process32FirstW(ql, address, params):
+    return 0x1
+
+
+# BOOL Process32NextW(
+#   HANDLE            hSnapshot,
+#   LPPROCESSENTRY32W lppe
+# );
+@winapi(cc=STDCALL, params={
+    "hSnapshot": HANDLE,
+    "lppe": POINTER
+
+})
+def hook_Process32NextW(ql, address, params):
+    # Return True if more process, 0 else
+    global COUNTER_PROCESS32_NEXT
```

Comment:
try to avoid global. we gonna bump into issue sooner or later.

Reply:
Understood, but I thought that was a bad practice insert it inside the class Qiling.
Suggestions on how to implement this concept?

Reply:
qiling class could be better. depends on how ofen u use it. or some class among there few windows API

Reply:
COUNTER_PROCESS32_NEXT use use in this function ? why do we need a global ?

Reply:
Meh, for now i think we can always return 1 safely.
I will just set a TODO to remember to change the function if we want a more complex implementation, but in this case we must implement correctly a snapshot handle (and the function createToolHelp32Snapshot) that i thought was not necessary.
If you agree with the simpler solution, I will change the functions and push, otherwise if we want to implement the handle, there is a lot of work to do.

Reply:
Should be fine. My only issue is the global var. maybe you can move in the class.

Reply:
The problem is that is only used in this function. If we do this with every variable that every hooks need, the class Qiling will explode with variables that have to use if not for a single hook. I don't think is a good practice either.

Reply:
if only this function so we dont need a global ?

Reply:
I mean yeah, but there is another way to implement a counter of how many times a function has been called, without use a global?

Reply:
we need to find a way or else if we decided to move to unittest, this global will screw things up. maybe we can implement a counter function with ql.counter['fuction_name'] ?

Reply:
I think this is a better way to implement it. Let me know how you prefer, if you want to think about it, for now I can just return 0 .

Reply:
make a function, i got a feeling we might need that for future for sure.

Reply:
296eca3 should resolve it